hack/gen-resources/generators/cluster_generator.go

```diff
@@ -176,70 +176,78 @@ func (cg *ClusterGenerator) retrieveClusterUri(namespace, releaseSuffix string)
 }
 
 func (cg *ClusterGenerator) Generate(opts *util.GenerateOpts) error {
-	for i := 1; i <= opts.ClusterOpts.Samples; i++ {
-		log.Printf("Generate cluster #%v of #%v", i, opts.ClusterOpts.Samples)
+	log.Printf("Excute in parallel with %v", opts.ClusterOpts.Concurrency)
 
-		namespace := opts.ClusterOpts.NamespacePrefix + "-" + util.GetRandomString()
+	wg := util.New(opts.ClusterOpts.Concurrency)
+	for l := 1; l <= opts.ClusterOpts.Samples; l++ {
+		wg.Add()
+		generateLoop := func(i int) error {
+			defer wg.Done()
+			log.Printf("Generate cluster #%v of #%v", i, opts.ClusterOpts.Samples)
 
-		log.Printf("Namespace is %s", namespace)
+			namespace := opts.ClusterOpts.NamespacePrefix + "-" + util.GetRandomString()
 
-		releaseSuffix := util.GetRandomString()
+			log.Printf("Namespace is %s", namespace)
 
-		log.Printf("Release suffix is %s", namespace)
+			releaseSuffix := util.GetRandomString()
 
-		err := cg.installVCluster(opts, namespace, POD_PREFIX+"-"+releaseSuffix)
-		if err != nil {
-			log.Printf("Skip cluster installation due error %v", err.Error())
-			continue
-		}
+			log.Printf("Release suffix is %s", namespace)
 
-		log.Print("Get cluster credentials")
-		caData, cert, key, err := cg.getClusterCredentials(namespace, releaseSuffix)
+			err := cg.installVCluster(opts, namespace, POD_PREFIX+"-"+releaseSuffix)
+			if err != nil {
+				log.Printf("Skip cluster installation due error %v", err.Error())
```

Comment:
It was with "continue" before, what is going to happens if vcluster cant be installed?